jkq/ddsim/unitarysimulator.py

```diff
@@ -0,0 +1,164 @@
+"""Backend for DDSIM Unitary Simulator."""
+
+import logging
+import uuid
+import time
+from math import log2, sqrt
+import warnings
+from typing import Union, List
+
+from qiskit.providers import BackendV1, Options
+from qiskit import QiskitError, QuantumCircuit
+from qiskit.providers.models import QasmBackendConfiguration
+from qiskit.qobj import QasmQobjExperiment, Qobj, QasmQobj, PulseQobj
+from qiskit.result import Result
+from qiskit.compiler import assemble
+from qiskit.utils.multiprocessing import local_hardware_info
+
+from .jkqjob import JKQJob
+from .jkqerror import JKQSimulatorError
+from jkq import ddsim
+
+import numpy as np
+
+logger = logging.getLogger(__name__)
+
+
+class UnitarySimulator(BackendV1):
+    """Decision diagram-based unitary simulator."""
+
+    def __init__(self, configuration=None, provider=None, **fields):
+        conf = {
+            'backend_name': 'unitary_simulator',
+            'backend_version': ddsim.__version__,
+            'n_qubits': min(24, int(log2(sqrt(local_hardware_info()['memory'] * (1024 ** 3) / 16)))),
+            'url': 'https://github.com/iic-jku/ddsim',
+            'simulator': True,
+            'local': True,
+            'conditional': False,
+            'open_pulse': False,
+            'memory': False,
+            'max_shots': 1000000000,
```

Comment:
Do `max_shots` make sense for the unitary simulator?

Reply:
Not really, but Qiskit specifies it for theirs so I figured it would make sense to do the same.